File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int count = 0;
    public int solution(int[] numbers, int target) {
        dfs(numbers, 0, 0, target);
        return count;
    }

    public void dfs(int[] numbers, int depth, int answer, int target) {
        if(depth == numbers.length) {
            if(answer == target) count++;
            return;
        }
        for (int i : new int[]{1, -1}) {
            dfs(numbers, depth + 1, answer + numbers[depth] * i, target);
        }
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
from itertools import combinations

def solution(numbers, target):
    answer = 0
    num_sum = sum(numbers)
    for i in range(len(numbers)):
        new_nums = combinations(numbers, i+1)
        for nums in new_nums:
            num = num_sum + sum([-2 * n for n in nums])
            if target == num:
                answer += 1
    return answer